src/test/java/com/orbitz/consul/BaseIntegrationTest.java

```diff
@@ -18,4 +24,16 @@ public static void beforeClass() {
                 .withReadTimeoutMillis(Duration.ofSeconds(2).toMillis())
                 .build();
     }
+
+    @After
```

Comment:
This is called after each test; but you do not clean the list.
If test class gets bigger, you will try to deregister services that no longer exists after every test.

Reply:
fixed